test/functional/wallet_labels.py

```diff
@@ -83,8 +83,14 @@ def run_test(self):
             label.add_receive_address(address)
             label.verify(node)
 
+        # Check listlabels when passing 'purpose'
+        node2_addr = self.nodes[1].getnewaddress()
+        node.setlabel(node2_addr, "node2_addr")
+        assert_equal(node.listlabels(purpose="send"), ["node2_addr"])
```

Comment:
TEST ACK, but It is not clear to me why node2_addr is a `send` address, I tried to replicate it in regtest but all my new addresses are `receive` addresses, maybe is worth clarifying it.

Reply:
`node = self.nodes[0]` -> so, `node` refers to `self.nodes[0]`.

Here we are getting a new address from `self.nodes[1]`, so it will receive btc using this address. For this reason, it can be used by `self.nodes[0]` to send Bitcoins.

So, `node2_addr` for `self.nodes[1]` is "receive" and for `self.nodes[0]` is "send".